platform/admin/core/admin-core-appservice/src/test/java/org/codice/ddf/admin/application/service/command/ProfileInstallCommandTest.java

```diff
@@ -0,0 +1,262 @@
+        doThrow(new ResolutionException("")).when(featuresService).installFeature("badFeature", NO_AUTO_REFRESH);
+        profileInstallCommand.doExecute(applicationService, featuresService, bundleService);
+        verify(featuresService, times(2)).installFeature(anyString(), NO_AUTO_REFRESH);
+    }
+
+    @Test(expected = ResolutionException.class)
+    public void testExtraProfileInvalidFeatureUninstall() throws Exception {
+        profileInstallCommand.profileName = "invalidFeatureUninstall";
+        doThrow(new ResolutionException("")).when(featuresService).uninstallFeature("badFeature", "0.0.0", NO_AUTO_REFRESH);
+        profileInstallCommand.doExecute(applicationService, featuresService, bundleService);
+        verify(featuresService, times(2)).uninstallFeature(anyString(), NO_AUTO_REFRESH);
+    }
+
+    @Test(expected = BundleException.class)
+    public void testExtraProfileInvalidStopBundle() throws Exception {
+        profileInstallCommand.profileName = "invalidStopBundles";
+        Bundle badBundle = mock(Bundle.class);
+        doThrow(new BundleException("")).when(badBundle).stop();
+        when(bundleService.getBundle("badBundle")).thenReturn(badBundle);
+        profileInstallCommand.doExecute(applicationService, featuresService, bundleService);
+        verify(bundleService, times(2)).getBundle(anyString());
+    }
+
+    @Test
+    public void testExtraProfilesWhenNotExists() throws Exception {
+        profileInstallCommand.profileName = "nonExistent";
+        profileInstallCommand.doExecute(applicationService, featuresService, bundleService);
+        verifyMocksNoOp();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testProfileStartsFolderTraversal() throws Exception {
+        profileInstallCommand.profileName = "../testProfile";
+        profileInstallCommand.doExecute(applicationService, featuresService, bundleService);
+        verifyMocksNoOp();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testProfileStartsSlash() throws Exception {
+        profileInstallCommand.profileName = "/testProfile";
+        profileInstallCommand.doExecute(applicationService, featuresService, bundleService);
+        verifyMocksNoOp();
+    }
+
+    @Test
+    public void testValidRegularProfile() throws Exception {
+        profileInstallCommand.profileName = "standard";
+        Feature feature = createMockFeature("standard");
+        List<Dependency> deps = Arrays.asList(createMockDependency("app1"),
+                createMockDependency("app2"), createMockDependency("app3"));
+        when(feature.getDependencies()).thenReturn(deps);
+        when(applicationService.getInstallationProfiles()).thenReturn(Collections.singletonList(feature));
+        profileInstallCommand.doExecute(applicationService, featuresService, bundleService);
+        verify(applicationService, times(3)).startApplication(anyString());
+    }
+
```

Comment:
❗️We need to test the following paths in `uninstallInstallerModule()`:
* `admin-post-install-modules` is already installed
* `admin-modules-installer` feature is installed
* `featuresService.installFeature("admin-post-install-modules", ...)` throws an exception
* `featuresService.uninstallFeature("admin-modules-installer", ...)` throws an exception.